homeassistant/components/sleepiq/config_flow.py

```diff
@@ -41,17 +43,14 @@ async def async_step_user(
             await self.async_set_unique_id(user_input[CONF_USERNAME].lower())
             self._abort_if_unique_id_configured()
 
-            login_error = await self.hass.async_add_executor_job(
-                try_connection, user_input
-            )
-            if not login_error:
+            try:
+                await try_connection(self.hass, user_input)
                 return self.async_create_entry(
                     title=user_input[CONF_USERNAME], data=user_input
                 )
```

Comment:
Please move this out of the `try` block since it cannot throw any of the exceptions behind caught

Like an `else` block